pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -792,49 +790,47 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 		pod := c.pods.getPodByIP(proxyIP)
 		if workload, f := c.workloadInstancesByIP[proxyIP]; f {
 			var err error
-			out, err = c.hydrateWorkloadInstance(workload)
+			out, err := c.hydrateWorkloadInstance(workload)
 			if err != nil {
 				log.Warnf("hydrateWorkloadInstance for %v failed: %v", proxy.ID, err)
 			}
+			return out, nil
 		} else if pod != nil {
-			// for split horizon EDS k8s multi cluster, in case there are pods of the same ip across clusters,
-			// which can happen when multi clusters using same pod cidr.
-			// As we have proxy Network meta, compare it with the network which endpoint belongs to,
-			// if they are not same, ignore the pod, because the pod is in another cluster.
-			if proxy.Metadata.Network != c.endpointNetwork(proxyIP) {
-				return out, nil
+			if !c.isControllerForProxy(proxy) {
```

Comment:
actually both works well, it is the meshnetworks is invalid from your configutation.